### DISCUSSION TITLE
Wrong input: Not existing vector name

### DISCUSSION THREAD

**Discussion by kientv** [qdrant/qdrant#3213] [Q&A] [answered]
I created collection name by this code:

> qdrantClient.create_collection(
>         collection_name=collection_name,
>         vectors_config=VectorParams(
>             distance=rest.Distance.COSINE,
>             size=1536,
>         ),
>     )

But error as below when upsert:

>   File "/home/chat-langchain/main.py", line 80, in ingress
>     ingest_docs(llm, collection_name, source_type, url, max_depth)
>   File "/home/chat-langchain/ingest.py", line 135, in ingest_docs
>     indexing_stats = index(
>                      ^^^^^^
>   File "/home/chat-langchain/_index.py", line 154, in index
>     vector_store.add_documents(docs_to_index, ids=uids)
>   File "/home/chat-langchain/venv/lib/python3.11/site-packages/langchain_core/vectorstores.py", line 122, in add_documents
>     return self.add_texts(texts, metadatas, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/chat-langchain/venv/lib/python3.11/site-packages/langchain/vectorstores/qdrant.py", line 181, in add_texts
>     self.client.upsert(
>   File "/home/chat-langchain/venv/lib/python3.11/site-packages/qdrant_client/qdrant_client.py", line 987, in upsert
>     return self._client.upsert(
>            ^^^^^^^^^^^^^^^^^^^^
>   File "/home/chat-langchain/venv/lib/python3.11/site-packages/qdrant_client/qdrant_remote.py", line 1300, in upsert
>     http_result = self.openapi_client.points_api.upsert_points(
>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/chat-langchain/venv/lib/python3.11/site-packages/qdrant_client/http/api/points_api.py", line 1439, in upsert_points
>     return self._build_for_upsert_points(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/chat-langchain/venv/lib/python3.11/site-packages/qdrant_client/http/api/points_api.py", line 738, in _build_for_upsert_points
>     return self.api_client.request(
>            ^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/chat-langchain/venv/lib/python3.11/site-packages/qdrant_client/http/api_client.py", line 74, in request
>     return self.send(request, type_)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/chat-langchain/venv/lib/python3.11/site-packages/qdrant_client/http/api_client.py", line 97, in send
>     raise UnexpectedResponse.for_response(response)
> qdrant_client.http.exceptions.UnexpectedResponse: Unexpected Response: 400 (Bad Request)
> Raw response content:
> b'{"status":{"error":"Wrong input: Not existing vector name error: "},"time":0.064342823}'
> 

Do you know how to fix it.

Thanks in advanced.

**Comment by timvisee** [comment]
Are you sure that you're inserting in this very same collection? Things look about right.

It errors because it cannot insert a vector with name `""`, but the collection you're creating does have that name (`""` the default for an unnamed vector).

**Comment by kientv** [comment]
> Are you sure that you're inserting in this very same collection? Things look about right.
> 
> It errors because it cannot insert a vector with name `""`, but the collection you're creating does have that name (`""` the default for an unnamed vector).

I'm using same vector name:

`debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"Wrong input: Not existing vector name error:  custom_vector", grpc_status:3, created_time:"2023-12-12T17:26:11.522274035+07:00"}"`

My code like this:

>      vectorstore = Qdrant(
>             client=qdrantClient,
>             collection_name=collection_name,
>             embeddings=vertex_embedding,
>             metadata_payload_key="tags",
>             content_payload_key="content",
>             vector_name="custom_vector",
>         )

>     qdrantClient.create_collection(
>         collection_name=collection_name,
>         vectors_config=VectorParams(
>             distance=rest.Distance.COSINE,
>             size=768,
>         ),
>     )

**Reply by timvisee** [reply, marked as answer]
Based on the above it looks like you're:
- creating a collection with no named vectors
- are upserting vectors with the name `custom-vector`

The schema during creation and the vectors you upsert must match. Qdrant will not create new vector names dynamically.

In this example you can see how you'd create a collection with named vectors. In your case you likely just want one named vector (`custom-vector`): https://qdrant.tech/documentation/concepts/collections/#collection-with-multiple-vectors

After doing that I assume the above issue will be resolved.

**Comment by kientv** [comment]
Thank you for your support, Now it work with this code:

>      qdrantClient.create_collection(
>         collection_name=collection_name,
>         vectors_config= {
>             "custom_vector": rest.VectorParams(
>                 distance=rest.Distance.COSINE,
>                 size=768,
>             ),
>         }
>     )
